cmd/config.go

```diff
@@ -12,32 +12,41 @@ import (
 
 const defaultConfigPath = ".circleci/config.yml"
 
+// Path to the config.yml file to operate on.
+// Used to for compatibility with `circleci config validate --path`
+var configPath string
+
 func newConfigCommand() *cobra.Command {
 	configCmd := &cobra.Command{
 		Use:   "config",
 		Short: "Operate on build config files",
 	}
 
 	collapseCommand := &cobra.Command{
-		Use:   "collapse [path]",
+		Use:   "collapse PATH",
 		Short: "Collapse your CircleCI configuration to a single file",
 		RunE:  collapseConfig,
 		Args:  cobra.MaximumNArgs(1),
 	}
 
 	validateCommand := &cobra.Command{
-		Use:     "validate [config.yml]",
+		Use:     "validate PATH (use \"-\" for STDIN)",
 		Aliases: []string{"check"},
 		Short:   "Check that the config file is well formed.",
 		RunE:    validateConfig,
 		Args:    cobra.MaximumNArgs(1),
 	}
+	validateCommand.PersistentFlags().StringVarP(&configPath, "path", "p", ".circleci/config.yml", "path to build config")
```

Comment:
The arg in build agent is not -p - it’s something else, -c I think. I assume that’s what you meant here?

Reply:
There is a constant at the top of this file that defines the default path.

Reply:
True, I actually copied this from earlier in the history of this repo when it was originally added but looks like we use `-c` or `--config`.